cistem/protocols/protocol_ts_resample.py

```diff
@@ -0,0 +1,229 @@
+        # Is the input a volume [YES]                        : NO
+        # New X-Size [464]                                   : 928
+        # New Y-Size [464]                                   : 928
+
+
+        # You need a params to belong to a section:
+        form.addSection(label=Message.LABEL_INPUT)
+        form.addParam('inputSetOfTiltSeries', PointerParam,
+                        pointerClass='SetOfTiltSeries',
+                        allowsNull=False,
+                        label='Input tilt series')
+
+        form.addParam('newXsize', IntParam,
+                        default=512,
+                        label='New X-Size',
+                        help='Volume will be rescaled to this size in X dimension (voxels)')
+        form.addParam('newYsize', IntParam,
+                        default=512,
+                        label='New Y-Size',
+                        help='Volume will be rescaled to this size in Y dimension (voxels)')
+
+    # -------------------------- INSERT steps functions -----------------------
+    def _insertAllSteps(self):
+
+        for ts in self.inputSetOfTiltSeries.get():
+
+            self._insertFunctionStep(self.runTsResample,
+                        ts.getObjId())
+
+            self._insertFunctionStep(self.createOutputStep,
+                        ts.getObjId())
+        
+        self._insertFunctionStep('closeStreamStep')
+
+    def runTsResample(self, tsObjId):
+
+        prog = Plugin.getProgram('resample')
+
+        ts = self.inputSetOfTiltSeries.get()[tsObjId]
+        firstItem = ts.getFirstItem()
+        tsFile = firstItem.getFileName()
+
+        tsBaseName = removeBaseExt(tsFile)
+        tsExt = getExt(tsFile)
+        paramDict = {}
+        paramDict['tsFile'] = tsFile
+        paramDict['tsOutName'] = self.getWorkingDir() + '/' + OUTPUT_DIR + '/' + \
+            tsBaseName + '_resampled' + tsExt
+        
+        paramDict['newXsize'] = self.newXsize
+        paramDict['newYsize'] = self.newYsize
+
+        # Arguments to the resample command defined in the plugin initialization:
+        args = """   << eof
+%(tsFile)s
+%(tsOutName)s
+NO
+%(newXsize)d
+%(newYsize)d
+eof\n
+"""
+        self.runJob(prog, args % paramDict)
+
+        self.tsList.append(paramDict['tsOutName'])
+
+    def createOutputStep(self, tsObjId):
+        ts = self.inputSetOfTiltSeries.get()[tsObjId]
+
+        firstItem = ts.getFirstItem()
+        tsFile = firstItem.getFileName()
+        tsBaseName = removeBaseExt(tsFile)
+        tsExt = getExt(tsFile)
+
+        tsOutName = self.getWorkingDir() + '/' + OUTPUT_DIR + '/' + \
+            tsBaseName + '_resampled' + tsExt
+
+        tsId = ts.getTsId()
+
+        oldSamplingRate = ts.getSamplingRate()
+        oldXsize = ts.getDim()[0]
+        self.samplingRate = oldSamplingRate * oldXsize / float(self.newXsize)
+
+        outputSetOfTiltSeries = self.getOutputSetOfTiltSeries()
+
+        newTs = TiltSeries(tsId=tsId)
```

Comment:
below is a lot of code that can be simplified. If your goal is only to change pixel size and filename compared to the input TS, look again at the warp deconv ts protocol.